mypy/checkexpr.py

```diff
@@ -2674,6 +2674,12 @@ def overload_arg_similarity(actual: Type, formal: Type) -> int:
             return overload_arg_similarity(actual.ret_type, formal.item)
         else:
             return 0
+    if isinstance(actual, TypedDictType):
+        if isinstance(formal, TypedDictType):
+            # Don't support overloading based on the keys or value types of a TypedDict since
+            # that would be complicated and probably only marginally useful.
```

Comment:
Since you felt compelled to add a comment about that, how sure are you? What are the symptoms when someone tries to overload on two different TypedDict types? Will calls with any kind of dict then just fail with a vague error? Does this deserves opening a tracker issue to debate it further?